File: src/main/webapp/src/service/api/permission-group-service.ts
import { request } from '../http'

export const PermissionGroupService = {
  getListModule() {
    return request.Get<Service.ResponseResult<any>>('/api/permission-group/get-list-module')
  },
  getListPermissionGroup() {
    return request.Get<Service.ResponseResult<any>>('/api/permission-group/get-list-permission-group')
  },
  getListPermissionByGroup(permissionGroupId: string) {
    return request.Get<Service.ResponseResult<any>>(`/api/permission-group/get-list-permission-by-group/${permissionGroupId}`)
  },
  createOrUpdatePermissionGroup(data: any) {
    return request.Post<Service.ResponseResult<any>>('/api/permission-group/create-or-update', data)
  },
}

File: src/main/java/vubq/warehouse_management/VT_EcoStorage/services/PermissionGroupService.java
package vubq.warehouse_management.VT_EcoStorage.services;

import org.springframework.data.domain.Page;
import vubq.warehouse_management.VT_EcoStorage.dtos.PermissionGroupDto;
import vubq.warehouse_management.VT_EcoStorage.dtos.requests.PermissionGroupRequest;
import vubq.warehouse_management.VT_EcoStorage.dtos.responses.ModuleResponse;
import vubq.warehouse_management.VT_EcoStorage.dtos.responses.PermissionGroupResponse;

import java.util.List;

public interface PermissionGroupService {

    Page<PermissionGroupResponse> getAllPermissionGroups();

    List<ModuleResponse> getListModule();

    List<PermissionGroupDto> getListPermissionGroup();

    PermissionGroupResponse getListPermissionByGroup(String permissionGroupId);

    PermissionGroupResponse createOrUpdatePermissionGroup(PermissionGroupRequest request);
}
